### ISSUE TITLE
Panic upon syntax highlighting

### ISSUE THREAD

**Issue by Omnikar** [helix-editor/helix#381]
### Reproduction steps
1. Open a file with a highlighted language.
2. Type text that should be highlighted.

As soon as text should be highlighted, Helix panics.

Example of one such panic message:

```
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Byte index out of bounds: byte index 6, Rope/RopeSlice byte length 3'
```

 

### Environment

- Platform: macOS
- Helix version: 0.2.0

**Comment by cessen** [comment]
I get it on my machine as well, but it's a little more specific than the repro steps above: it has to be a sufficiently *short* file.  Longer files don't trigger it, which is an interesting clue.

**Comment by cessen** [comment]
Looks like it's because of my fix in #377 for the other issue.  Fix one bug only to cause another...

**Comment by cessen** [comment]
Found and fixed the bug.  Just waiting for a review and merge of the PR now.

It was a silly mistake a made in my previous fix.
